statistics/selectivity.go

```diff
@@ -456,3 +456,25 @@ func GetUsableSetsByGreedy(nodes []*StatsNode) (newBlocks []*StatsNode) {
 	}
 	return
 }
+
+// FindPrefixOfIndexByCol will find columns in index by checking the unique id or the virtual expression.
+// So it will return at once no matching column is found.
+func FindPrefixOfIndexByCol(cols []*expression.Column, idxColIDs []int64, cachedPath *planutil.AccessPath) []*expression.Column {
+	if cachedPath != nil {
+		idxCols := cachedPath.IdxCols
+		retCols := make([]*expression.Column, 0, len(idxCols))
+	idLoop:
+		for _, idCol := range idxCols {
```

Comment:
@wjhuang2016 Is it possible to reduce the for loop?

Reply:
No idea.